x/ethbridge/keeper/ethereumLockBurnSequence.go

```diff
@@ -36,10 +36,42 @@ func (k Keeper) GetEthereumLockBurnSequence(ctx sdk.Context, networkDescriptor o
 	return binary.BigEndian.Uint64(bz)
 }
 
-// getEthereumLockBurnSequencePrefix return storage prefix
-func (k Keeper) getEthereumLockBurnSequencePrefix(networkDescriptor oracletypes.NetworkDescriptor, valAccount sdk.ValAddress) []byte {
+// GetEthereumLockBurnSequencePrefix return storage prefix
+func (k Keeper) GetEthereumLockBurnSequencePrefix(networkDescriptor oracletypes.NetworkDescriptor, valAccount sdk.ValAddress) []byte {
 	bytebuf := bytes.NewBuffer([]byte{})
 	_ = binary.Write(bytebuf, binary.BigEndian, networkDescriptor)
 	tmpKey := append(types.EthereumLockBurnSequencePrefix, bytebuf.Bytes()...)
 	return append(tmpKey, valAccount...)
 }
+
+func (k Keeper) getEthereumLockBurnSequenceIterator(ctx sdk.Context) sdk.Iterator {
+	store := ctx.KVStore(k.storeKey)
+	return sdk.KVStorePrefixIterator(store, types.EthereumLockBurnSequencePrefix)
+}
+
+// GetEthereumLockBurnSequences get all sequences from keeper
+func (k Keeper) GetEthereumLockBurnSequences(ctx sdk.Context) map[string]uint64 {
+	sequences := make(map[string]uint64)
+	iterator := k.getEthereumLockBurnSequenceIterator(ctx)
+	defer func(iterator sdk.Iterator) {
+		err := iterator.Close()
+		if err != nil {
+			panic(err)
+		}
+	}(iterator)
+	for ; iterator.Valid(); iterator.Next() {
+		key := iterator.Key()
+		value := iterator.Value()
+		sequences[string(key)] = binary.BigEndian.Uint64(value)
+	}
+	return sequences
+}
+
+// SetSequenceViaRawKey used in import sequence from genesis
+func (k Keeper) SetSequenceViaRawKey(ctx sdk.Context, key []byte, newSequence uint64) {
```

Comment:
This looks very error prone. 
I suggest write a function to extract NetworkDescriptor and ValidatorAddress, and use those to call `SetEthereumLockBurnSequence`

Reply:
Ok, will add extract method.

Reply:
Are these comments still current?

Reply:
I implemented a new method DecodeKey to extract NetworkDescriptor and ValidatorAddress according to comment. please check.